bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiAudioSink.java

```diff
@@ -89,20 +89,31 @@ public void process(AudioStream audioStream)
                 String url = ((URLAudioStream) audioStream).getURL();
                 logger.trace("Processing audioStream URL {} of format {}.", url, format);
                 handler.playURI(new StringType(url));
```

Comment:
The problem is that handler.playURI is apparently asynchronous.
I believe you are exactly in the case of the Sonos binding, with something asynchronous for URLAudioStream and something synchronous for all other audio streams.

Reply:
OK, I get it.
I've done the same kind of change you made on Sonos.

Reply:
Fine. I should have added similar good comments in the Sonos binding ;)à